gps_common/msg/GPSStatus.msg

```diff
@@ -17,6 +17,8 @@ int16 STATUS_FIX=0       # Normal fix
 int16 STATUS_SBAS_FIX=1  # Fixed using a satellite-based augmentation system
 int16 STATUS_GBAS_FIX=2  #          or a ground-based augmentation system
 int16 STATUS_DGPS_FIX=18 # Fixed with DGPS
+int16 STATUS_RTK_FIX=19  # Real-Time Kinematic, fixed integers
```

Comment:
Adding these explicit enum values is a reasonable suggested change; however, the md5 sum will change and that will likely cause unexpected compatibility issues for anyone who isn't building this package from source. I notice that the existing constants (e.g. GPSStatus::STATUS_DGPS_FIX) are not referenced by name (in `client.cpp` at least), so there isn't much pressing need (just looking at the present changes) to include named constants in the message file. Let me suggest instead adding these two new lines as comments instead. Unfortunately, they won't show up in `rosmsg show gps_common/GPSStatus`, but they will appear in the message definitions and online documentation (e.g. http://docs.ros.org/melodic/api/gps_common/html/msg/GPSStatus.html). More importantly, though, adding them as comments won't change the md5 sum. You can verify this with `rosmsg md5 gps_common/GPSStatus`

Reply:
client.cpp can not use the constants
because of the global STATUS_DGPS_FIX
from gpsd. When adding both as
comments people won't be able to use
GPSStatus::STATUS_RTK_FIX and will
fall back to using the number values
which I would really like to avoid.

Shouldn't the compatibility only be an issue when mixing different ROS distributions?

However there is a third option: I revert both constants. I do not need STATUS_RTK_FIX/STATUS_RTK_FLT that much since I trust the covariance values more then the status value. The "p->status & STATUS_FIX" is the important change for me.

Reply:
> client.cpp can not use the constants
> because of the global STATUS_DGPS_FIX
> from gpsd. When adding both as
> comments people won't be able to use
> GPSStatus::STATUS_RTK_FIX and will
> fallback to the number values
> which I would really like to avoid.

I agree with this statement, but I think that compatibility is a bigger issue than the use of hard-coded numbers. Comments in place of constants in the message definition is not ideal, but not the worst solution either.  

> Shouldn't the compatibility only be an issue when mixing different ROS distributions?

As far as I know, all supported distributions are currently using master, so the change would affect all distributions. It may be reasonable to make changes for a future (as yet unreleased) distribution, but that's a different story. The main compatibility issue will be that any mixing of the new `GPSStatus` (and also `GPSFix`) messages with the old definition will result in problems -- like failure to pass messages -- you'll probably see `WARN` level log messages in these cases. There are a few ways that mixing would/could occur, but basically any nodes that use the new definition (including those that are built against the new definition, or that would dynamically update, like python nodes) couldn't communicate with those that don't (any nodes not built against the new definition, which would include any binaries distributed via apt that have not been updated with the new message definitions). Additionally, this change also causes headaches for analyzing bagged data that includes GPSStatus and GPSFix messages of both types. 

> However there is a third option: I revert both constants. I do not need STATUS_RTK_FIX/STATUS_RTK_FLT that much since I trust the covariance values more then the status value. The "p->status & STATUS_FIX" is the important change for me.

I would definitely recommend prioritizing not changing the message definition over everything else here, so this sounds like it may be your best option. @pjreed, any thoughts, since you're probably the ultimate decider here?

Reply:
done. STATUS_RTK_FIX/STATUS_RTK_FLT removed